test/connectivity/exported_service_test.go

```diff
@@ -37,90 +40,128 @@ var _ = Describe("Exported Service", func() {
-		_, err := kubectlWithConfig(sharedServiceClusterKubeConfig,
+		By("declaring intent to export the service to the client cluster")
+		_, err := kubectlWithConfig(servicesClusterKubeConfig,
 			"apply", "-f", filepath.Join("..", "..", "manifests", "example", "nginx", "exported_http_proxy.yaml"))
 		Expect(err).NotTo(HaveOccurred())
 
-		By("validating it can connect to the shared service on the workload cluster")
+		By("validating it can connect to the published service on the client cluster")
 		Eventually(func() (string, error) {
-			output, err := kubectlWithConfig(workloadClusterKubeConfig,
+			output, err := kubectlWithConfig(clientClusterKubeConfig,
 				"run", "nginx-test", "-i", "--rm", "--image=curlimages/curl", "--restart=Never", "--",
 				"curl", "-v", "-k", "--connect-timeout", curlConnectTimeoutInSeconds, "https://nginx.xcc.test")
 			return string(output), err
 		}, kubectlTimeout, kubectlInterval).Should(And(
 			ContainSubstring("HTTP/2 200"),
-			ContainSubstring("Thank you for using nginx."),
+			ContainSubstring(fmt.Sprintf("x-cluster: %s", servicesClusterID)),
 		))
 
-		By("remove intent to export the shared service to the workload cluster")
-		_, err = kubectlWithConfig(sharedServiceClusterKubeConfig,
+		By("remove intent to export the published service to the client cluster")
+		_, err = kubectlWithConfig(servicesClusterKubeConfig,
 			"-n", "nginx-test", "label", "httpproxy", "nginx-service-ingress", "connectivity.tanzu.vmware.com/export-")
 		Expect(err).NotTo(HaveOccurred())
 
-		By("validating it cannot connect to the shared service on the workload cluster")
+		By("validating it cannot connect to the published service on the client cluster")
 		Eventually(func() string {
-			output, _ := kubectlWithConfig(workloadClusterKubeConfig,
+			output, _ := kubectlWithConfig(clientClusterKubeConfig,
 				"run", "nginx-test", "-i", "--rm", "--image=curlimages/curl", "--restart=Never", "--",
 				"curl", "-s", "-S", "--stderr", "-", "-k", "--connect-timeout", curlConnectTimeoutInSeconds, "https://nginx.xcc.test")
 			return string(output)
 		}, kubectlTimeout, kubectlInterval).Should(ContainSubstring("Could not resolve host"))
-	})
+
+		_, err = kubectlWithConfig(servicesClusterKubeConfig,
+			"delete", "namespace", "nginx-test")
+		Expect(err).NotTo(HaveOccurred())
+
+		By("ensuring cross-cluster-connectivity managed resources are cleaned up")
+		Eventually(func() (string, error) {
+			output, err := kubectlWithConfig(servicesClusterKubeConfig,
+				"get", "servicerecord", "-n", connectivityNamespace)
+			return string(output), err
+		}, kubectlTimeout, kubectlInterval).ShouldNot(ContainSubstring("nginx.xcc.test"))
+
+		Eventually(func() (string, error) {
+			output, err := kubectlWithConfig(clientClusterKubeConfig,
+				"get", "servicerecord", "-n", connectivityNamespace)
+			return string(output), err
+		}, kubectlTimeout, kubectlInterval).ShouldNot(ContainSubstring("nginx.xcc.test"))
```

Comment:
Noob question. If the tests are running in parallel, won't there be a chance of test pollution? Because we aren't only looking at either Imported or Exported ServiceRecords. So won't there be a chance that an exported service record shows up, when we expect no imported service record?

Reply:
I think you are right. I think I might move this into an AfterSuite then. I don't know how important I think it is that each test sees a clean slate in the cross-cluster-connectivity namespace, especially if I want to run tests in parallel. Perhaps we still care that no lingering files are left around, but we do that after the entire test suite is complete.